src/applications/microservices/payforadoption-go/config.go

```diff
@@ -104,5 +111,33 @@ func fetchConfigFromParameterStore(ctx context.Context, cfg payforadoption.Confi
 
 // Call aws secrets manager and return parsed sql server query str
 func getRDSConnectionString(ctx context.Context, cfg payforadoption.Config) (string, error) {
+	if refreshManager != nil && !refreshManager.shouldRefreshSecret() {
+		if secret, ok := refreshManager.getCachedSecret(); ok {
+			InfoWithTrace(ctx, "Using cached database secret\n")
+			var dbConfig payforadoption.DatabaseConfig
+			if err := json.Unmarshal([]byte(secret), &dbConfig); err == nil {
+				u := &url.URL{
+					Scheme: dbConfig.Engine,
+					User:   url.UserPassword(dbConfig.Username, dbConfig.Password),
+					Host:   fmt.Sprintf("%s:%d", dbConfig.Host, dbConfig.Port),
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Potential log injection detected. Ensure all untrusted input is properly sanitized before logging. Use parameterized logging or validate input against an allow list to prevent log injection vulnerabilities. Consider using a dedicated logging library&#39;s built-in sanitization features when available. Learn more - https://cwe.mitre.org/data/definitions/117.html